orderbook/src/database/presignatures.rs

```diff
@@ -0,0 +1,375 @@
+use crate::conversions::h160_from_vec;
+use crate::database::Postgres;
+use anyhow::{anyhow, Context, Result};
+use ethcontract::H160;
+use futures::{
+    future,
+    stream::{BoxStream, StreamExt, TryStreamExt},
+};
+use model::{order::OrderUid, presignature::PreSignature};
+use std::convert::TryInto;
+
+#[async_trait::async_trait]
+pub trait PreSignatureRetrieving: Send + Sync {
+    fn presignatures<'a>(
+        &'a self,
+        filter: &'a PreSignatureFilter,
+    ) -> BoxStream<'a, Result<PreSignature>>;
+}
```

Comment:
I just merged in another PR a change that makes the orders and trades return Vec directly instead of Stream. We should do the same here and then it can have a metric in Instrumented too.